File: Backend/ApiUniversidad/ApiUniversidad/Models/Curso.cs
using System;
using System.Collections.Generic;

namespace ApiUniversidad.Models;

public partial class Curso
{
    public Guid Id { get; set; }

    public string Nombre { get; set; } = null!;

    public DateTime FechaCreacion { get; set; }

    public string Horarios { get; set; } = null!;

    public Guid IdCarrera { get; set; }

    public virtual ICollection<AlumnosPorCurso> AlumnosPorCursos { get; set; } = new List<AlumnosPorCurso>();

    public virtual ICollection<DocentesPorCurso> DocentesPorCursos { get; set; } = new List<DocentesPorCurso>();

    public virtual CarrerasUniversidad IdCarreraNavigation { get; set; } = null!;
}

File: Frontend/ApiUniversidad/UniversidadAngular/src/app/models/curso.ts
import { AlumnosPorCurso } from "./alumnos-por-curso";
import { CarrerasUniversidad } from "./carreras-universidad";
import { DocentesPorCurso } from "./docentes-por-curso";

export class Curso {
    id?: string;
    nombre: string = '';
    fechaCreacion: Date = new Date();
    horarios: string = '';
    idCarrera: string = '';
    nombreCarrera?: string; // Para DTO del backend
    alumnosPorCurso?: AlumnosPorCurso[];
    docentesPorCurso?: DocentesPorCurso[];
    idCarreraNavigation?: CarrerasUniversidad;
}
